src/services/tokens.ts

```diff
@@ -35,7 +33,11 @@ export const getValidJwt = async (authorizationToken: string, logEvent: ILogEven
   }
 
   logEvent.email = username;
+  logEvent.roles = (decoded.payload as JwtPayload).roles;
```

Comment:
This duplicates what happens in `roles.ts` but is needed here as well to ensure that roles are still logged in the event of other failures and the logs are consistent.

we could remove this line and just have 'undefined' in that scenario. Do we care what the roles of a user are, if that is not the reason the token has failed validation?

Reply:
I think keeping the roles in there is better for the sake of consistency. Are the roles guaranteed to be populated? If so then I do not see much downside to keeping it.

Reply:
We can't guarantee anything unfortunately, as an invalid JWT could theoretically contain anything

Reply:
Perhaps put a null guard in there just to be safe?

Reply:
I'm not sure if we can guard on it, as if it _is_ null, then we probably want to log out that it is, as that will inform what the issue is when debugging this log.